corehq/messaging/scheduling/views.py

```diff
@@ -492,3 +494,119 @@ def schedule_form(self):
+
+    @method_decorator(_requires_new_reminder_framework())
+    @method_decorator(requires_privilege_with_fallback(privileges.OUTBOUND_SMS))
+    @method_decorator(require_permission(Permissions.edit_data))
+    @use_datatables
+    def dispatch(self, *args, **kwargs):
+        return super(ConditionalAlertListView, self).dispatch(*args, **kwargs)
+
+
+class CreateConditionalAlertView(BaseMessagingSectionView, AsyncHandlerMixin):
+    urlname = 'create_conditional_alert'
+    page_title = _('New Conditional Message')
+    template_name = 'scheduling/conditional_alert.html'
+    async_handlers = [MessagingRecipientHandler]
+
+    @method_decorator(_requires_new_reminder_framework())
+    @method_decorator(requires_privilege_with_fallback(privileges.OUTBOUND_SMS))
+    @method_decorator(require_permission(Permissions.edit_data))
+    @use_jquery_ui
+    @use_timepicker
+    @use_select2
+    def dispatch(self, *args, **kwargs):
+        return super(CreateConditionalAlertView, self).dispatch(*args, **kwargs)
+
+    @property
+    def parent_pages(self):
+        return [
+            {
+                'title': ConditionalAlertListView.page_title,
+                'url': reverse(ConditionalAlertListView.urlname, args=[self.domain]),
+            },
+        ]
+
+    @property
+    def page_context(self):
+        return {
+            'basic_info_form': self.basic_info_form,
+            'criteria_form': self.criteria_form,
+        }
+
+    @property
+    def initial_rule(self):
+        return None
+
+    @cached_property
+    def read_only_mode(self):
+        return (
+            not self.is_system_admin and
+            self.criteria_form.requires_system_admin_to_edit
+        )
+
+    @cached_property
+    def is_system_admin(self):
+        return self.request.couch_user.is_superuser
+
+    @cached_property
+    def basic_info_form(self):
+        if self.request.method == 'POST':
+            return ConditionalAlertForm(self.domain, self.request.POST)
+
+        return ConditionalAlertForm(self.domain)
+
+    @cached_property
+    def criteria_form(self):
+        kwargs = {
+            'rule': self.initial_rule,
+            'is_system_admin': self.is_system_admin,
+        }
+
+        if self.request.method == 'POST':
+            return ConditionalAlertCriteriaForm(self.domain, self.request.POST, **kwargs)
+
+        return ConditionalAlertCriteriaForm(self.domain, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        basic_info_form_valid = self.basic_info_form.is_valid()
+        criteria_form_valid = self.criteria_form.is_valid()
+
+        if self.read_only_mode:
+            # Don't allow making changes to rules that have custom
+            # criteria/actions unless the user has permission to
+            return HttpResponseBadRequest()
+
+        if basic_info_form_valid and criteria_form_valid:
+            if not self.is_system_admin and self.criteria_form.requires_system_admin_to_save:
```

Comment:
nit: could this be changed to `if self.read_only_mode:`?

Reply:
the logic in `self.read_only_mode` is slightly different - it's checking for `requires_system_admin_to_edit`, which inspects the existing rule (only relevant when editing) and decides whether it should be shown as read only in the UI.

the check that's happening here is for `requires_system_admin_to_save`, and it's checking right before save if there are options which have been enabled which require a superuser. we shouldn't ever hit this because the UI restricts it anyway, but since anything can be posted it's needed.

Reply:
Ah, I missed that subtle difference. Thanks for the clarification.